lib/types/types.go

```diff
@@ -23,9 +23,63 @@ package types
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
+	"reflect"
 	"time"
+
+	null "gopkg.in/guregu/null.v3"
 )
 
+func NullDecoder(f reflect.Type, t reflect.Type, data interface{}) (interface{}, error) {
+	typeFrom := f.String()
+	typeTo := t.String()
+
+	expectedType := ""
+	switch typeTo {
+	case "null.String":
+		if typeFrom == reflect.String.String() {
+			return null.StringFrom(data.(string)), nil
+		}
+		expectedType = reflect.String.String()
+	case "null.Bool":
+		if typeFrom == reflect.Bool.String() {
+			return null.BoolFrom(data.(bool)), nil
+		}
+		expectedType = reflect.Bool.String()
+	case "null.Int":
+		if typeFrom == reflect.Int.String() {
+			return null.IntFrom(int64(data.(int))), nil
+		}
+		if typeFrom == reflect.Int32.String() {
+			return null.IntFrom(int64(data.(int32))), nil
+		}
+		if typeFrom == reflect.Int64.String() {
+			return null.IntFrom(data.(int64)), nil
+		}
+		expectedType = reflect.Int.String()
+	case "null.Float":
+		if typeFrom == reflect.Float32.String() {
+			return null.FloatFrom(float64(data.(float32))), nil
+		}
+		if typeFrom == reflect.Float64.String() {
+			return null.FloatFrom(data.(float64)), nil
+		}
+		expectedType = reflect.Float32.String() + " or " + reflect.Float64.String()
+	case "types.NullDuration":
+		if typeFrom == reflect.String.String() {
+			var d NullDuration
+			err := d.UnmarshalText([]byte(data.(string)))
+			return d, err
+		}
```

Comment:
I think you should set `expectedType` here?

Reply:
yes, we should 👍